### PR TITLE
fix(collapsible): fix keyboard events

### PR DESCRIPTION
**Pull request by iropolo**
# Fix collapsible keyaboard events

## :hammer_and_wrench: Type Of Change

- [x] Fix

## :book: Description

The purpose of the change is fix a bug explained in [DT-763](https://dialpad.atlassian.net/browse/DT-763)

## :bulb: Context

We moved directly the `@click` event into the button and remove parents events.
With this we avoid side effects between these.

We also update the test file reflecting this change.

## :pencil: Checklist

- [x] I have reviewed my changes
- [x] I have added tests
- [x] I have validated components keyboard navigation
- [x] I have considered the performance impact of my change

## :camera: Screenshots / GIFs

image

## :link: Sources

[DT-763](https://dialpad.atlassian.net/browse/DT-763)

### CODE REVIEW

File: components/collapsible/collapsible.test.js [hunk 1]
```diff
@@ -1,40 +1,40 @@
 import { assert } from 'chai';
 import { config, mount } from '@vue/test-utils';
-import DtCollpasible from './collapsible.vue';
```
Comment (braddialpad) [inline components/collapsible/collapsible.test.js:3]: lol

File: components/collapsible/collapsible.vue [hunk 2]
```diff
@@ -105,7 +103,7 @@ export default {
      */
     anchorText: {
       type: String,
-      default: null,
+      default: 'Button',
```
Comment (iropolo) [inline components/collapsible/collapsible.vue]: This was required for `axe-core accessibility`

_Personal Opinion_ I understand this as mandatory, we should have anchor with text as a default value.

Before it was added by the slot, but we removed the slot override to show the default button instead.

Here we have two options, give the text value by props, or just adding it as a default value.

Comment (braddialpad) [reply components/collapsible/collapsible.vue]: I think the correct way to fix this is with a custom validator, that validates for this:

> anchorText and anchor slot content cannot both be falsy

You can see an example in combobox.vue, `validateEmptyListProps()`

**Review by braddialpad — COMMENTED** [review]
Congrats on your first PR!

File: components/collapsible/collapsible.vue (hunk 2 above)
Comment (iropolo) [reply components/collapsible/collapsible.vue]: Completely agree :)

**Review by braddialpad — APPROVED** [review]
Thanks!

**Pull request merged**